db/model/generator.js

```diff
@@ -154,6 +155,45 @@ module.exports = function generator(seq, dataTypes) {
           override: true,
         });
       },
+
+      /**
+       * 1. validate the collectors field: if succeed, save the collectors in temo var for
+       *  attaching to generator. if fail, abort the POST operation
+       * 2. create the generator
+       * 3. add the saved collectors (if any)
+       */
+      createWithCollectors(requestBody, user) {
+        const options = {};
+        let collectors; // will be populated with actual collectors
+        options.where = common.whereClauseForNameInArr(requestBody.collectors || []);
+        return new seq.Promise((resolve, reject) =>
+          seq.models.Collector.findAll(options)
+          .then((_collectors) => {
+
+            /*
+             * If requestBody does not have a collectors field, OR
+             * if the number of collectors in requestBody MATCH the
+             * GET result, order the collectors AND create the generator.
+             * Else throw error since there are collectors that don't exist.
+             */
+            if (!requestBody.collectors ||
+              (_collectors.length === requestBody.collectors.length)) {
```

Comment:
what do we do when "requestBody.collectors" contains duplicate collector names?

Reply:
adding a test for that

Reply:
ok added the test to show the result of POSTing with duplicate collectors is 400